pkg/ingester-rf1/ingester.go

```diff
@@ -260,15 +286,20 @@ func New(cfg Config, clientConfig client.Config, store Store, limits Limits, con
 		tailersQuit:     make(chan struct{}),
 		metrics:         metrics,
 		//flushOnShutdownSwitch: &OnceSwitch{},
-		terminateOnShutdown: false,
-		//streamRateCalculator:  NewStreamRateCalculator(),
+		terminateOnShutdown:  false,
+		streamRateCalculator: NewStreamRateCalculator(),
 		writeLogManager:      writefailures.NewManager(logger, registerer, writeFailuresCfg, configs, "ingester_rf1"),
 		customStreamsTracker: customStreamsTracker,
 		readRing:             readRing,
+		flushCtx: &flushCtx{
+			lock:            &sync.RWMutex{},
+			flushDone:       make(chan struct{}),
+			newCtxAvailable: make(chan struct{}),
+			segmentWriter:   segmentWriter,
+		},
```

Comment:
Interesting choice here. 

I thought we would need multiple flushes routine per ingester to be able to flush faster, right now if a single flush is getting slow down (happens) all other requests will be affected.

Not a problem though let's put it to the test in dev.

Reply:
We do have multiple flush routines - the idea was to have one "active" flushCtx at a time, which all Pushs are Appending to.
Every time the timer fires, it will swap the flushCtx for a new one (new wal segment, etc.) as fast as possible - it's just a variable assignment - and then hand off the old flushCtx to be flushed async.
I fully expect flushing to storage to be the slowest part of this, so I'm reusing the existing flush loops. We can have multiple flushes happening in parallel, and only the Pushes that appended to that flush would be waiting. New Pushes would be unaffected as they'd be Appending to a completely different flushCtx.

Definitely open to suggestions if you have better ideas though!